crates/cli/src/bin/wasm-bindgen-test-runner/server.rs

```diff
@@ -45,36 +46,129 @@ pub fn spawn(
-    ));
+    if worker {
+        let worker_script = format!(
+            r#"
+                importScripts("{0}.js");
+
+                const wrap = method => {{
+                    self.console[method] = function (...args) {{
+                        postMessage(["__wbgtest_" + method, args]);
+                    }};
+                }};
+
+
+                self.__wbg_test_invoke = f => f();
+
+                wrap("debug");
+                wrap("log");
+                wrap("info");
+                wrap("warn");
+                wrap("error");
+
+                async function run_in_worker(tests) {{
+                    const wbg = await wasm_bindgen("./{0}_bg.wasm");
+                    const t = self;
+                    const cx = new wasm_bindgen.WasmBindgenTestContext();
+                    cx.args({1:?});
+                    await cx.run(tests.map(s => wbg[s]));
+                }}
+
+                onmessage = function(e) {{
+                    let tests = e.data;
+                    run_in_worker(tests);
+                }}
+            "#,
+            module, args,
+        );
+
+        let worker_js_path = tmpdir.join("worker.js");
+        fs::write(&worker_js_path, worker_script).context("failed to write JS file")?;
+
+        js_to_execute.push_str(&format!(
+            r#"
+                // Now that we've gotten to the point where JS is executing, update our
+                // status text as at this point we should be asynchronously fetching the
+                // wasm module.
+                document.getElementById('output').textContent = "Loading wasm module...";
+                const worker = new Worker("worker.js", {{module: {1}}});
+
+                worker.addEventListener("message", function(e) {{
+                    console.log(e);
+                    // Checking the whether the message is from wasm_bindgen_test
+                    if(
+                        e.data &&
+                        Array.isArray(e.data) &&
+                        e.data[0] &&
+                        typeof e.data[0] == "string" &&
+                        e.data[0].slice(0,10)=="__wbgtest_"
+                    ) {{
+                        const method = e.data[0].slice(10);
+                        const args = e.data.slice(1);
+
+                        if (
+                            method == "log" || method == "warn" ||
+                            method == "warn" || method == "info" ||
+                            method == "debug"
+                        ) {{
+                            console[method].apply(undefined, args[0]);
+                        }} else if (method == "start") {{
+                            document.getElementById("output").textContent = "";
+                        }} else if (method == "output") {{
+                            document.getElementById("output").textContent += args[0] + "\n";
+                        }}
+                    }}
+                }});
```

Comment:
This is more complicated than it needs to be. You don't really need to have a wasm module in both the main thread and the worker.

Instead, you should be able to have only the wasm module in the worker, have it keep track of what the contents of the output `<pre>` should be, and then post a message back to the main thread with the updated contents every time it changes.